kairon/nlu/classifiers/llm.py

```
@@ -272,15 +271,16 @@ def persist(self) -> None:
+            data_file = os.path.join(model_path, file_name + "_data.json")
+
             if self.vector and self.data:
                 create_directory_for_file(model_path)
                 faiss.write_index(
-                    self.vector, os.path.join(model_path, vector_file_name)
+                    self.vector, vector_file
                 )
-                io_utils.json_pickle(
-                    os.path.join(model_path, data_file_name), self.data
+                json.dump(
+                    self.data, open(data_file, "w"), escape_forward_slashes=True
```

Ruff (0.8.2)

Use a context manager for opening files

(SIM115)